From 46c60959feebf8bc82d79a144e67ca85ab6e6afa Mon Sep 17 00:00:00 2001
From: Stefan Roese <sr@denx.de>
Date: Tue, 18 Dec 2018 09:26:32 +0100
Subject: [PATCH] MIPS: ralink: dts: gardena_smart_gateway_mt7688: Add
 bootcounter DT node

Add the U-Boot bootcounter DT node, so that the driver can register on
it.

Signed-off-by: Stefan Roese <sr@denx.de>
Upstream-Status: Inappropriate [for U-Boot bootcounter driver]
(cherry picked from commit e339d5255ea3b8c9fafe4fb64d4aa12edda7c3af)
---
 arch/mips/boot/dts/ralink/gardena_smart_gateway_mt7688.dts | 7 +++++++
 arch/mips/boot/dts/ralink/mt7628a.dtsi                     | 2 +-
 2 files changed, 8 insertions(+), 1 deletion(-)

diff --git a/arch/mips/boot/dts/ralink/gardena_smart_gateway_mt7688.dts b/arch/mips/boot/dts/ralink/gardena_smart_gateway_mt7688.dts
index 7eb927a30722..4929286c08be 100644
--- a/arch/mips/boot/dts/ralink/gardena_smart_gateway_mt7688.dts
+++ b/arch/mips/boot/dts/ralink/gardena_smart_gateway_mt7688.dts
@@ -212,3 +212,10 @@ &wmac {
 	status = "okay";
 	mediatek,mtd-eeprom = <&factory 0x0000>;
 };
+
+&palmbus {
+	bootcount@0x6c {
+		compatible = "uboot,bootcount";
+		reg = <0x0000006c 0x4>;
+	};
+};
diff --git a/arch/mips/boot/dts/ralink/mt7628a.dtsi b/arch/mips/boot/dts/ralink/mt7628a.dtsi
index 5b90d5cf09ac..3fd05e967466 100644
--- a/arch/mips/boot/dts/ralink/mt7628a.dtsi
+++ b/arch/mips/boot/dts/ralink/mt7628a.dtsi
@@ -24,7 +24,7 @@ cpuintc: interrupt-controller {
 		compatible = "mti,cpu-interrupt-controller";
 	};
 
-	palmbus@10000000 {
+	palmbus: palmbus@10000000 {
 		compatible = "palmbus";
 		reg = <0x10000000 0x200000>;
 		ranges = <0x0 0x10000000 0x1FFFFF>;
-- 
2.43.0

